tern/analyze/docker/dockerfile.py

```diff
@@ -118,6 +126,38 @@ def parse_from_image(dfobj):
     return image_list
 
 
+def expand_from_images(dfobj):
+    '''Replace all parent_image values with their full tag@sha:digest value.
+    Update the structure dictionary with the same information.
+    dfobj: the Dockerfile object created using get_dockerfile_obj'''
+    # update parent_images
+    images = parse_from_image(dfobj)
+    for i, img in enumerate(images):
+        dfobj.parent_images[i] = find_image_digest(img['name'], img['tag'])
+    # update structure
+    for i, command_dict in enumerate(dfobj.structure):
+        if command_dict['instruction'] == 'FROM':
+            dfobj.structure[i]['content'] = 'FROM ' + find_image_digest(
+                command_dict['value'].split(tag_separator, 2)[0],
+                command_dict['value'].split(tag_separator, 2)[1])
+            dfobj.structure[i]['value'] = find_image_digest(
+                command_dict['value'].split(tag_separator, 2)[0],
+                command_dict['value'].split(tag_separator, 2)[1])
+
+
+def find_image_digest(image_name, image_tag):
+    '''Given an image name and tag from the FROM instruction, return the
+    digest information of the unique image in 'tag@sha:digest' format.'''
+    container.check_docker_setup()
+    image_tag_string = image_name + tag_separator + image_tag
+    logger.debug("Attempting to pull image \"%s\"", image_tag_string)
+    try:
+        return container.client.images.pull(image_tag_string).attrs[
+            'RepoDigests'][0]
+    except docker.errors.ImageNotFound:
+        logger.warning("No such image: \"%s\"", image_tag_string)
```

Comment:
I wonder if this should live in `container.py` instead of `dockerfile.py`.

Reply:
Sure, I can move it.